presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -14,14 +14,20 @@
 package com.facebook.presto.server;
 
 import com.google.common.base.StandardSystemProperty;
+import com.sun.management.UnixOperatingSystemMXBean;
 import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
 import org.joda.time.DateTime;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
 import java.nio.ByteOrder;
 
 final class PrestoSystemRequirements
 {
+
+    private static final int MIN_FILE_DESCRIPTORS = 10_000;
```

Comment:
ok, I guess your mileage really may vary. Reading the value from a config file seems more convenient than having to set it at the command line. What if the check is disabled by default but enabled when set > 0? We could commit a reasonable default in the configs so going forward people would pick it up.

Also wrt naming ```min-file-descriptors``` sounds like you're configuring how many to allow, vs configuring what the check threshold is. We should name to convey the latter.

In addition to this, another helpful enhancement would be to try to find where the exception gets thrown when exceeded and add more messaging to it suggesting checking the ulimit. Now sure how feasible it would be to reliably do that.